### DISCUSSION TITLE
How to add file extension in url

### DISCUSSION THREAD

**Discussion by Montekkundan** [arthelokyo/astrowind#65] [Q&A]
I was wondering how can i make the website url look like this:

`www.website.com/test.md` or `www.website.com/test.mdx `

instead of `www.website.com/test`

**Comment by prototypa** [comment]
Hi @Montekkundan 

Here I bring you a solution. There are a few other ways to achieve what you need. Please tell me if it works for you.

1- Go to this file https://github.com/onwidget/astrowind/blob/main/src/utils/posts.js#L3 and replace:

Before:
```js
const ID = file.split('/').pop().split('.').shift();
```

After:
```js
const ID = file.split('/').pop();
```

<br />

2- Go to this file https://github.com/onwidget/astrowind/blob/main/src/utils/permalinks.js#L21 and replace:

Before:
```js
export const cleanSlug = (text) => slugify(trimSlash(text));
```

After:
```js
export const cleanSlug = (text) => slugify(trimSlash(text), { custom: ['.'] })
```

**Comment by prototypa** [comment]
You know @Montekkundan, with your question I've been thinking: _"what if we add a settings to [`config.mjs`](https://github.com/onwidget/astrowind/blob/main/src/config.mjs) with the slug pattern?"_

Ex:

```javascript
export const BLOG = {
  // (..)

  post: {
    disabled: false,
    pathname: '', // empty for /some-post, value for /pathname/some-post
    slugPattern: "%post_id%.md", // NEW PROPOSED SETTING
  },

  // (...)
};
```

Some of the _frontmatter_ variables could be available as variables, example: **_%title%_** (with slugification), **_%category%_** (first if it is a list), **_%author%_** (first author) and some others generated as **_%year%_**, **_%monthnum%_**, **_%day%_**, **_%hour%_**, **_%minute%_**, **_%second%_**. All very similar to what Wordpress brings to configure the permalinks

The default slug pattern would be **_%post_id%_**

---

It's just an idea, we have to analyze it better so as not to complicate the template too much, what do you think?

**Reply by Montekkundan** [reply]
the idea is great! will first try the steps you listed above and report back! busy right now with packing 😅 (shifting home). Will report as soon as possible! 

Thanks for helping out
